src/hooks/api/authenticated-query.ts

```diff
@@ -151,10 +152,36 @@ const prefetchAuthenticatedQuery = async <TData>({
 const useAuthenticatedMutation = ({ mutationFn, ...configuration }) => {
   const router = useRouter();
   const headers = useHeaders();
+  const queryClient = useQueryClient();
 
   const { removeAuthenticationCookies } = useCookiesWithOptions();
 
   const innerMutationFn = useCallback(async (variables) => {
+    try {
+      const mutationCache = queryClient.getMutationCache().getAll();
+
+      const cacheForMutationKey = mutationCache.filter((mutation) => {
+        return (
+          typeof configuration.mutationKey !== 'undefined' &&
+          mutation.options.mutationKey === configuration.mutationKey
+        );
+      });
```

Comment:
I believe we can do `getMutationCache().find({ mutationKey: configuration.mutationKey })` here to retrieve a potential mutation from cache, judging from the TS types. This would return a single mutation object which would also prevent having to do the array stuff

Reply:
Yes we also saw the function, but it isn't documented though. We suspect it maybe is a internal function.

https://tanstack.com/query/v3/docs/react/reference/MutationCache

Reply:
Yeah we can use the `find` function, since it will possibly be a bit more performant

Reply:
@Anahkiasen Could you take another look after my latest changes?

Reply:
We still have to use the `findAll` instead of `find` because we always want the second last mutation. Because the mutation that gets executed gets added to the cache first before this check happens. Thus comparing against `mutations[0]` is comparing against itself.